deeplabcut/live/dlc_live.py

```diff
@@ -32,131 +30,180 @@ class DLCLive(object):
+        """
+
+        if os.path.basename(self.cfg_file) == "config.yaml":
+            # find the pose_config.yaml file
+            self.cfg = auxiliaryfunctions.read_config(self.cfg_file)
+
+            # warn if overriding project_path
+            # override in case project was copied from somewhere else without changing config
+            if os.path.dirname(self.cfg_file) != self.cfg['project_path']:
+                Warning('config.yaml not in project_path, using the directory of config.yaml')
+                self.cfg["project_path"] = os.path.dirname(self.cfg_file)
+
+            # get model folder
+            trainFraction = self.cfg['TrainingFraction'][self.trainingsetindex]
+            self.model_folder = os.path.join(self.cfg["project_path"],
+                                       str(auxiliaryfunctions.GetModelFolder(
+                                           trainFraction, self.shuffle, self.cfg))
+
+                                       )
+            path_test_config = Path(self.model_folder) / 'test' / 'pose_cfg.yaml'
+            try:
+                self.dlc_cfg = load_config(str(path_test_config))
+            except FileNotFoundError:
+                raise FileNotFoundError("It seems the model for shuffle {} and trainFraction {} does not exist.".format(
+                    self.shuffle, trainFraction)
+                )
+
+        else:
+            ValueError('config given at instantiation must be the config.yaml file at the root of the project directory!')
+
+
+        # gpu selection
+        if self.gputouse:
+            os.environ['CUDA_VISIBLE_DEVICES'] = str(self.gputouse)
+
+        if self.iteration:
             self.cfg['iteration'] = self.iteration
-        trainFraction = self.cfg['TrainingFraction'][self.trainingsetindex]
 
         if self.cropping is not None:
             self.cfg['cropping']=True
             self.cfg['x1'], self.cfg['x2'], self.cfg['y1'], self.cfg['y2'] = self.cropping
             print("Overwriting cropping parameters:", self.cropping)
 
-        self.cfg["project_path"] = os.path.dirname(config)
-        modelfolder = os.path.join(self.cfg["project_path"],str(auxiliaryfunctions.GetModelFolder(trainFraction, self.shuffle, self.cfg)))
-        path_test_config = Path(modelfolder) / 'test' / 'pose_cfg.yaml'
-        try:
-            self.dlc_cfg = load_config(str(path_test_config))
-        except FileNotFoundError:
-            raise FileNotFoundError("It seems the model for shuffle %s and trainFraction %s does not exist."%(self.shuffle, trainFraction))
-
         # Check which snapshots are available and sort them by # iterations
         try:
-          Snapshots = np.array([fn.split('.')[0]for fn in os.listdir(os.path.join(modelfolder , 'train'))if "index" in fn])
+            snapshots = np.array([fn.split('.')[0]for fn in os.listdir(os.path.join(self.model_folder , 'train'))if "index" in fn])
```

Comment:
check is backwards compt; snapshots vs. Shapshots

Reply:
completely fine to revert, 2-spacing should -> 4-spacing

Reply:
I'm leaving this in for now, it's functional